### PR TITLE
Optional use QueueSubscribe for NATS events source

### LINKED ISSUE #2936: Use `QueueSubscribe` for NATS events source?
Right now NATS event source uses `subscribe` and because of that the has to use the [Active/Passive strategy](https://argoproj.github.io/argo-events/eventsources/ha/#active-passive) for HA. 

Could we not switch to using [Queue Subscriptions](https://docs.nats.io/using-nats/developer/receiving/queues) giving us the ability to have Active/Active HA?

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by sysr9**
resolves #2936 

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

adds queue parameter to NATS EventSource to implement queue-based architecture

### CODE REVIEW

**Comment by praveenperera** [comment]
Wouldn't it make more sense to make this the default, and switch the active-active failover?

**Comment by praveenperera** [comment]
@sysr9 makes sense, if we are using the QueueGroup how can we make it use active-active HA mode? Or should that be a different PR?

File: pkg/apis/eventsource/v1alpha1/types.go [hunk 1]
```diff
@@ -631,6 +631,10 @@ type NATSEventsSource struct {
 	// Filter
 	// +optional
 	Filter *EventSourceFilter `json:"filter,omitempty" protobuf:"bytes,8,opt,name=filter"`
+	// Queue is the name of the queue group to subscribe as if specified. Uses QueueSubscribe
+	// logic to subscribe as queue group. If the queue is empty, uses default Subscribe logic.
+	// +optional
+	Queue string `json:"queue" protobuf:"bytes,9,opt,name=queue"`
```
Comment (whynowy) [inline pkg/apis/eventsource/v1alpha1/types.go:637]: Use `*string`, all the rest looks good to me, thanks!

**Review by whynowy — APPROVED** [review]
LGTM

**Pull request merged**